### PR TITLE
feat(token): add scoped API tokens and token scopes subcommand

### LINKED ISSUE #144: feat(token): add scopes support to API token commands
## Background

The server now supports scoped API tokens via a `scopes` field in `resource:action` format (e.g., `server:read`, `command:create`). Tokens default to `["*"]` (full access), but operators can now restrict tokens to only the permissions they actually need.

The CLI currently has no way to specify scopes when creating a token, does not display scopes in the token list, and provides no way to discover what scopes are available.

### 1. `token create` — add `--scopes` flag

```
alpacon token create --name ci-token --scopes "server:read,command:create"
alpacon token create --name deploy-token --scopes "*"
```

- Accept a comma-separated list of scope strings
- Pass them to `POST /api/tokens/` as the `scopes` field
- Interactive prompt should also ask for scopes (with a default of `*`)

### 2. `token ls` — add scopes column

Display the `scopes` field in the list output so operators can see at a glance what each token is allowed to do.

### 3. `token scopes` — new subcommand

```
alpacon token scopes
```

Calls `GET /api/tokens/scopes/` and prints the available scope resources and their actions, filtered to what the current user has RBAC permission to grant.

## API reference

| Method | Endpoint | Description |
|--------|----------|-------------|
| `POST` | `/api/tokens/` | `scopes` field now accepted |
| `GET` | `/api/tokens/scopes/` | List available scopes for current user |

## Acceptance criteria

- [ ] `token create --scopes "server:read"` creates a token with the given scopes
- [ ] `token create` interactive mode prompts for scopes
- [ ] `token ls` displays the scopes column
- [ ] `token scopes` lists available resources and their actions
- [ ] Omitting `--scopes` defaults to `["*"]` (backward compatible)

### PR DESCRIPTION
**Pull request by jisung-02**
Closes #144

## Summary

Adds scoped API-token support to the CLI so callers can mint least-privilege tokens for CI/CD and agents:

- `alpacon token create --scopes "server:read,command:create"` (wildcards `*`, `server:*`, `*:read` supported)
- `alpacon token ls` now includes a `Scopes` column
- New `alpacon token scopes` subcommand enumerates the scope resources, actions, and ACLs the current user is allowed to grant

## Changes

- `api/auth`
  - `APITokenRequest.Scopes` with `omitempty` (legacy callers unchanged on the wire)
  - `APITokenResponse.Scopes`, projected into `APITokenAttributes.Scopes` for table output
  - New `TokenScopesResponse` / `TokenScopeResource` / `TokenScopeAttributes`
  - New `GetTokenScopes()` hitting `/api/auth/tokens/scopes/`
- `cmd/token`
  - `token create --scopes` flag + interactive prompt (empty input falls back to server-side `*` default)
  - New `token scopes` subcommand
- `utils`
  - Extracted `SplitAndTrim` and reused it from `PromptForListInput`

## Presentation notes

Wildcard rows come back from the API layer with empty `Actions`. The table renderer in `token scopes` fills in `(matches all scopes)` only for table output, so `--output json` stays free of human-readable strings.

## Stack

golang

## Impact

- Depends on the server exposing `/api/auth/tokens/scopes/` and accepting `scopes` on `POST /api/auth/tokens/`. Older self-hosted servers will get an error from `token scopes`; `token create` without `--scopes` keeps working (field is `omitempty`).

## Checklist

- [x] Code follows project style guidelines (golangci-lint v2 clean)
- [x] Self-reviewed the code
- [x] No print/log debug statements left
- [x] No hardcoded secrets or credentials
- [x] Error handling is complete on every new code path
- [x] Tests cover edge cases: scopes omitempty on create, Scopes column population, wildcards + empty ACL in token scopes

## Test plan

- [x] `go test -race ./...`
- [x] `golangci-lint run ./...`
- [ ] Against staging: `alpacon token create --name ci-token --scopes "server:read,command:create"` and confirm the issued token can hit only those endpoints
- [ ] Against staging: `alpacon token create --name full --scopes "*"` issues a full-access token
- [ ] `alpacon token create` (interactive, empty scopes) falls back to server default
- [ ] `alpacon token ls` shows the new `Scopes` column
- [ ] `alpacon token scopes` lists wildcards + resources; `--output json` emits empty `actions` for wildcard rows

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -467,3 +467,19 @@ func CommandConfirm() bool {
...
+	if s == "" {
+		return nil
+	}
+	parts := strings.Split(s, sep)
+	result := make([]string, 0, len(parts))
+	for _, p := range parts {
+		if trimmed := strings.TrimSpace(p); trimmed != "" {
+			result = append(result, trimmed)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply utils/utils.go:483]: Fixed in b8efd2e...ba9d775 — `SplitAndTrim` now returns `nil` when the input yields no non-empty elements, matching the documented contract.

File: cmd/token/token_create.go [hunk 2]
```diff
@@ -11,14 +11,19 @@ var tokenCreateCmd = &cobra.Command{
 	Use:   "create",
 	Short: "Create a new api token",
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/token/token_create.go]: Fixed in b8efd2e — updated to "Create a new API token".

File: cmd/token/token_scopes.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import (
+	"github.com/alpacax/alpacon-cli/api/auth"
+	"github.com/alpacax/alpacon-cli/client"
+	"github.com/alpacax/alpacon-cli/utils"
+	"github.com/spf13/cobra"
+)
+
+var tokenScopesCmd = &cobra.Command{
+	Use:   "scopes",
+	Short: "List available scopes for api tokens",
```
[the remark this answers is not in the document]
Comment (jisung-02) [reply cmd/token/token_scopes.go]: Fixed in b8efd2e — updated to "List available scopes for API tokens".

**Review by eunyoung14 — APPROVED** [review]
